src/baldr/graphtile.cc

```diff
@@ -372,6 +372,7 @@ GraphId GraphTile::GetTileId(const std::string& fname) {
       break;
     }
   }
+  allowed.erase(static_cast<std::string::value_type>(filesystem::path::preferred_separator));
```

Comment:
after this point we no longer allow a separator as a valid character in the file path and we skip over them when we find them below